components/Layout.tsx

```diff
@@ -0,0 +1,140 @@
+    href: '/',
+  },
+  {
+    name: 'Homepage',
+    href: 'https://nation3.org',
+    icon: <HomeIcon className="h-5 w-5" />,
+  },
+  {
+    name: 'Docs',
+    href: 'https://docs.nation3.org/',
+    icon: <NewspaperIcon className="h-5 w-5" />,
+  },
+]
+
+export default function Layout({ children }: any) {
+  const router = useRouter()
+
+  const [nav, setNav] = useState(navigation)
+
+  return (
+    <div className="mx-auto bg-n3bg font-display">
+      <Script src="https://cdn.splitbee.io/sb.js" />
+      <div className="h-screen">
+        <div className="navbar bg-base-100 border-slate-100 border-b-2 py-0 pl-0 lg:hidden sticky z-10">
+          <div className="navbar-start border-slate-100 pl-0">
+            <div className="w-80 border-slate-100 py-4 box-content">
+              <div className="pl-6 pt-2 cursor-pointer">
+                <div className="flex-none hidden lg:block">
+                  <Link href="/" passHref>
+                    <a>
+                      <Image src={Logo} />
+                    </a>
+                  </Link>
+                </div>
+                <div className="flex-none lg:hidden">
+                  <label
+                    htmlFor="side-drawer"
+                    className="btn btn-square btn-ghost"
+                  >
+                    <Bars3Icon className="h-8 w-8" />
+                  </label>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div className="drawer drawer-mobile">
+          <input id="side-drawer" type="checkbox" className="drawer-toggle" />
+          <div className="drawer-content fixed top-0 left-0 right-0 bottom-0 lg:static pt-24 lg:pt-0 z-0 max-h-screen">
+            <div className="flex flex-col w-full h-full">
+              <div className="hero h-full">
+                <div className="hero-content">{children}</div>
+              </div>
+            </div>
+          </div>
+          <div className="drawer-side">
+            <label
+              htmlFor="side-drawer"
+              className="drawer-overlay z-10"
+            ></label>
+            <div className="bg-white w-80 flex flex-col justify-between pb-24 lg:pb-0 overflow-y-auto drop-shadow-md min-h-screen">
+              <div className="mt-6 py-4 hidden lg:block">
+                <div className="px-8 pt-2 cursor-pointer">
+                  <Link href="/" passHref>
+                    <a>
+                      <Image src={Logo} />
+                    </a>
+                  </Link>
+                </div>
+              </div>
+              <ul className="menu p-4 overflow-y-auto text-base-400 grow">
+                {nav.map((item: any) => (
+                  <li
+                    className="mt-1 relative py-2"
+                    onClick={() =>
+                      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
+                      (document.getElementById('side-drawer').checked = false)
```

Comment:
We should add a controller so that we do not need the getElementById and the site just reacts to the click. We shouldn't go wild with useEffect as it became a minfield on the app!